www/perl-lib/SMT/Utils.pm

```diff
@@ -127,6 +129,46 @@ sub db_connect
     return $dbh;
 }
 
+=item hasRegSharing()
+
+Check whether registration sharing is enabled or not.
+
+=cut
+
+sub hasRegSharing
+{
+    my $r = shift;
+
+    if (! defined $REGSHARING) {
+        $REGSHARING = 0;
+        my $cfg;
+        eval
+        {
+            $cfg = SMT::Utils::getSMTConfig();
+        };
+        if($@ || !defined $cfg)
+        {
+            if (! $r)
+            {
+                return;
+            }
+            $r->log_error("Cannot read the SMT configuration file: ".$@);
+            my $msg = 'SMT server is missconfigured. Please contact your '
+                . 'administrator.';
+            return http_fail($r, 500, $msg);
```

Comment:
OK, got it now. Thanks. For some reason I thought http_fail would basically set the status and then exit the current execution path. Obviously it has been a while since I looked at the implementation of http_fail.

Anyway, in order to set the proper status (500) I elected to stick with using http_fail, but I modified the logic.